libsolidity/experimental/codegen/IRGenerationContext.h

```diff
@@ -33,8 +33,12 @@ class Analysis;
 
 struct IRGenerationContext
 {
-	Analysis const& analysis;
-	TypeEnvironment const* env = nullptr;
+	struct QueuedFunction
+	{
+		FunctionDefinition const* function = nullptr;
+		Type type = std::monostate{};
+	};
+
```

Comment:
You're intentionally creating conflicts here, right :-)?

Reply:
Why were the fields not defined together though? I know this is a small thing, but it's just *so* frustrating to have to fish out the fields from between functions and type definitions.

Reply:
I'm not saying there's no logic behind this arrangement, but it's different from our usual pattern - and IMO the way we do this usually is better, because without it you have actually look closely to distinguish the variables from other things.

Also, the severity of this conflict is overstated too. The actually bad ones are the ones where something you relied on changes under you. This one is mildly annoying, but I doubt it would give much trouble anyone on the team. IMO it's worth it to fix these things because it's once and done.

Overall I really don't agree that it's better to do things like these along with changing the logic. It's what you usually have to do anyway when you work on top of something messy, but it's better when you can separate logic changes from refactors. It makes reviewing the actually important changes easier.

> It's not like anything close to the current version of this struct will remain long-term.

Well, ok. TBH I started assuming that this **is** the long-term version because we're going to merge it.

Reply:
The entire point in calling this experimental is being able to merge it *before* it's a final version, but to merge things that are not polished and that can be quickly iterated over.

And the point is not the severity of the conflict itself - the point is that changes to the base branch like this makes it virtually impossible to smoothly review dependent changes like e.g. https://github.com/ethereum/solidity/pull/14620 without prior rebases, so we need a constant loop of watching out for - or pinging people to - rebase PRs which overall delays things without any gain that would justify that.